packages/next/src/build/analysis/get-page-static-info.ts

```diff
@@ -45,11 +45,12 @@ export interface MiddlewareConfig {
 }
 
 export interface MiddlewareMatcher {
-  regexp: string
-  locale?: false
+  source: string
+  regexp?: string
```

Comment:
Isn't the `regexp?: string` still to much? The `regexp` key is not mentioned in the docs and I found this [Middleware](https://github.com/vercel/next.js/blob/v14.2.4/packages/next/src/lib/load-custom-routes.ts#L73) type which matches the expected layout from the docs and it also does not mention/have `regexp`

Reply:
It is [documented here](https://nextjs.org/docs/app/api-reference/file-conventions/middleware#config-object-optional) as:
> "A regular expression string that fine-tunes the matching based on the source. It provides additional control over which paths are included or excluded."

I do see that it's not documented in the pages router docs however, it may not do anything when used with the pages router, however these types make no distinction between the two router paradigms.

Reply:
Oh right, I totally overlooked that one. Sorry for the noise.